pandas/_libs/tslibs/period.pyx

```diff
@@ -1598,7 +1599,7 @@ cdef class _Period:
     def __richcmp__(self, other, op):
         if is_period_object(other):
             if other.freq != self.freq:
-                msg = DIFFERENT_FREQ.format(cls=type(self).__name__,
+                msg = DIFFERENT_FREQ(cls=type(self).__name__,
```

Comment:
You need an f string?

Reply:
same deal as abowe in frequencies.pyx file. DIFFERENT_FREQ is now a lambda function